frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -340,16 +346,14 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
             ? `Upload of dataset ${form.getFieldValue("name")} froze.`
             : `Uploading Dataset ${form.getFieldValue("name")}.`}
           <br />
-          {isRetrying ? "Retrying to continue the upload..." : null}
+          {isRetrying ? "Retrying to continue the upload …" : null}
           <br />
           <Progress
-            // Round to 1 digit after the comma. Don't show 100%, since there is
-            // some additional delay until the UI shows the next screen.
-            // Otherwise, the user might think that nothing will happen after 100%
-            // was reached.
-            percent={Math.min(99.9, Math.round(uploadProgress * 1000) / 10)}
+            // Round to 1 digit after the comma.
+            percent={Math.round(uploadProgress * 1000) / 10}
             status="active"
           />
+          {isFinishing ? <Spin tip="Processing uploaded files …" /> : null}
```

Comment:
When `isFinishing` is true, there is a progress bar **and** a spinner. I would only show one of them. One of the following options probably:
(1) Always show the progress bar. due to `status="active"`, the progress bar will have a slight animation even when 100% is reached. below the progress bar, "Processing uploaded files" would still be shown (so only the spinner would be removed).
(2) Hide the progress bar and show the spinner instead when `isFinishing` becomes true.

I'd do (1). What do you think?

Reply:
I tried (1) originally, but I feld like just adding a bit of text didn’t convey the change strongly enough. I’d actually vote for showing both, but I don’t have a strong opinion on this.

Reply:
ok, then let's go with your approach :)

Reply:
Thx for jumping in @philippotto. Sorry I am currently having pretty busy days.

Reply:
no worries!